### PR TITLE
Dynamic css loading (#1362)

### PR DESCRIPTION
**Pull request by mkeilman**
Notes:

Thought I'd start this while getting vtk into shape.  This excises warpvnd- and vtk- classes from ```sirepo.css``` into their own files and extends the code that loads .js.

### CODE REVIEW

File: sirepo/package_data/static/js/sirepo.js [hunk 1]
```diff
@@ -13,17 +13,33 @@ angular.element(document).ready(function() {
 
     function loadDynamicModule(src) {
         var d = $.Deferred();
-        var scriptTag = document.createElement('script');
-        scriptTag.type = 'text/javascript';
-        scriptTag.async = true;
-        scriptTag.onload = function() {
+        var tag;
+        var parent;
+        if(fileType(src) === '.css') {
```
[the remark this answers is not in the document]
Comment (mkeilman) [reply sirepo/package_data/static/js/sirepo.js]: I only use regex to please you, master...

File: sirepo/simulation_db.py [hunk 2]
```diff
@@ -965,7 +966,10 @@ def _file_list_in_schema(schema, file_type):
         str: combined list of local and external file paths
     """
     paths = []
+    pkdlog('checking schema {} for {}', schema, file_type)
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/simulation_db.py]: Debugging lines should be `pkdp`. `pkdlog` is for permanent logging. If you have `pkdp` calls in the code, builds will fail.

The documentation for [pkdebug](https://github.com/radiasoft/pykern/blob/master/pykern/pkdebug.py) is out of date. Something to fix.

File: sirepo/package_data/static/js/sirepo.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/package_data/static/js/sirepo.js]: nails, hammers, cranes, sawzall, etc. have different purposes. Perhaps [this DesignHint](https://github.com/radiasoft/pykern/wiki/DesignHints#learn-libraries-theres-an-api-for-that) should be called: there's a *right* API for that. And as with apps on a phone, no app is sometimes the best solution.

In this case, you are not really caring about files. What you want to be able to distinguish is between two very clear and controlled cases.  The simpler test is robust for this purpose. You don't need an assertion. You could have added a [dispatch](https://github.com/radiasoft/pykern/wiki/DesignHints#dispatch-dont-decorate) for the problem, too, and then a regular expression would be appropriate. However, given there are only two, not three cases, I think the conditional is appropriate.

File: sirepo/package_data/static/js/sirepo.js [hunk 3]
```diff
@@ -13,17 +13,33 @@ angular.element(document).ready(function() {
...
+            tag = document.createElement('link');
```
[the remark this answers is not in the document]
Comment (mkeilman) [reply sirepo/package_data/static/js/sirepo.js]: Something like:
```
    function loadDynamicModule(src) {
        var d = $.Deferred();
        addTag(src).onload = function () {
            d.resolve();
        };
        return d.promise();
    }

    function tagConfig(src) {
        if (src.endsWith('.css')) {
            return {
                'tagName': 'link',
                'parentTagName': 'head',
                'srcAttr': 'href',
                'attrs': {
                    'rel': 'stylesheet',
                },
            };
        }
        return {
            'tagName': 'script',
            'parentTagName': 'body',
            'srcAttr': 'src',
            'attrs': {
                'type': 'text/javascript',
                'async': true,
            },
        };
    }
    function addTag(src) {
        var cfg = tagConfig(src);
        var tag = document.createElement(cfg.tagName);
        tag[cfg.srcAttr] = src + SIREPO.SOURCE_CACHE_KEY;
        for(var attr in cfg.attrs) {
            tag[attr] = cfg.attrs[attr];
        }
        document.getElementsByTagName(cfg.parentTagName)[0].appendChild(tag);
        return tag;
    }

```
???  Or is it getting too precious?

**Comment by mkeilman** [comment]
I have dealt with that for the most part - e.g. shadow imports srw.css.  The main elegant thing I see used elsewhere is ```elegant-icon``` and its relations.  I think I'll pull those back into sirepo.css and rename them since they are relatively widely used

**Pull request merged**